demo/src/main/java/io/jafar/demo/Main.java

```diff
@@ -71,16 +76,33 @@ private static void runWithJafar(String[] args, LongAccumulator sum, AtomicInteg
         }
     }
 
-    private static void runWithJfr(String[] args, LongAccumulator sum, AtomicInteger cnt) throws IOException, CouldNotLoadRecordingException {
-        try (RecordingFile recording = new RecordingFile(Paths.get((args[1])))) {
+    private static void runWithJfr(File file, LongAccumulator sum, AtomicInteger cnt) throws IOException, CouldNotLoadRecordingException {
+        try (RecordingFile recording = new RecordingFile(file.toPath())) {
             while (recording.hasMoreEvents()) {
                 RecordedEvent e = recording.readEvent();
                 if (e.getEventType().getName().equals("jdk.ExecutionSample")) {
-                    sum.accumulate(e.getThread("sampledThread").getId());
+                    sum.accumulate(e.getThread("sampledThread").getJavaThreadId());
                     sum.accumulate(e.getStackTrace().getFrames().size());
                     cnt.incrementAndGet();
                 }
             }
         }
     }
+
+    private static void runWithJfrStream(File file, LongAccumulator sum, AtomicInteger cnt) throws IOException, CouldNotLoadRecordingException {
+        var es = EventStream.openFile(Paths.get(file.getPath()));
+        es.setReuse(true);
+        es.setOrdered(false);
+        es.onEvent("jdk.ExecutionSample", e -> {
+            sum.accumulate(e.getThread("sampledThread").getJavaThreadId());
+            sum.accumulate(e.getStackTrace().getFrames().size());
+            cnt.incrementAndGet();
+        });
+        es.start();
+        try {
```

Comment:
If start() is used, the parser will run in the same thread until there is no more data. No need to awaitTermination().